lib/features/modeling/behavior/TextAnnotationBehavior.js

```diff
@@ -12,6 +12,19 @@ export default function TextAnnotationBehavior(eventBus) {
 
   CommandInterceptor.call(this, eventBus);
 
+  // On Append, TextAnnotations will be created on the Root.
+  // The default for connections will create the connection in the parent of
+  // the source element, so we overwrite the parent here.
+  this.preExecute('connection.create', function(context) {
+    const { target } = context;
+
+    if (!is(target, 'bpmn:TextAnnotation')) {
```

Comment:
Would this work inside collapsed sub-processes? I'd rather change the logic here to explicitly target the highest up `process`, `collaboration` or `sub-process` (which can store the annotation).

Reply:
Yes, this works inside a collapsed subproces. The Annotation is always added the "plane" of the original target (cf. root element, cf. https://github.com/bpmn-io/bpmn-js/blob/b5bae70a64fde3c114e06b9a4850d6c7f6190295/lib/features/ordering/BpmnOrderingProvider.js#L146-L152

By picking the highest flownode container, we get the resize behavior you experienced before. I think it is easier and cleaner to render on the root, rather than hack the resize behavior to ignore annotations